### PR TITLE
statsd: use pystatsd pipeline to send batches

### PR DESCRIPTION
**Pull request by filippog**
the pipeline feature has been introduced in https://github.com/jsocol/pystatsd
since version 2.0

### CODE REVIEW

File: src/diamond/handler/stats_d.py [hunk 1]
```diff
@@ -135,6 +135,8 @@ def _send(self):
                     statsd.Counter(prefix, self.connection).increment(
                         name, value)
 
+        if hasattr(statsd, 'StatsClient'):
```
Comment (jaingaurav) [inline src/diamond/handler/stats_d.py]: What is the purpose of this check?

File: src/diamond/handler/stats_d.py [hunk 2]
```diff
@@ -152,7 +154,7 @@ def _connect(self):
             self.connection = statsd.StatsClient(
                 host=self.host,
                 port=self.port
-            )
+            ).pipeline()
```
Comment (jaingaurav) [inline src/diamond/handler/stats_d.py]: Is there documentation for this feature? Does this call need to be conditional for legacy support?

Comment (filippog) [inline src/diamond/handler/stats_d.py]: yup documented at http://statsd.readthedocs.org/en/v3.2.1/pipeline.html for https://pypi.python.org/pypi/statsd/ and introduced in version 2.0 according to jsocol/pystatsd@51d684e46d

afaict this collector supports both the above and https://pypi.python.org/pypi/python-statsd/1.7.2 with the former being actively maintained on github but not the latter, hence the conditional

File: src/diamond/handler/stats_d.py (hunk 1 above)
Comment (filippog) [inline src/diamond/handler/stats_d.py]: conditional to support both python-statsd and statsd, see also other comment

File: src/diamond/handler/stats_d.py (hunk 2 above)
Comment (jaingaurav) [inline src/diamond/handler/stats_d.py]: Is there a reason to support both versions of the library? Seems like the former offers a superior API.

Comment (filippog) [inline src/diamond/handler/stats_d.py]: besides backward compatibility? no idea if there is still a good reason nowadays, possibly not

**Comment by jaingaurav** [comment]
Thanks for the clarification. Perhaps in a future diamond version we can deprecate support for the older library

**Comment by filippog** [comment]
yup, I can followup with PRs to change python-statsd with statsd in tox/travis/etc

**Pull request merged**